File: app/src/main/java/com/icostel/arhitecturesample/di/modules/AppModule.java
package com.icostel.arhitecturesample.di.modules;

import android.app.Application;
import android.content.Context;

import com.bumptech.glide.annotation.GlideModule;

import dagger.Binds;
import dagger.Module;

@Module(includes = {
        ViewModelModule.class,
        ApiModule.class,
        HttpClientModule.class,
        DbModule.class}
)
public abstract class AppModule {

    @Binds
    abstract Context bindContext(Application application);
}

File: app/src/main/java/com/icostel/arhitecturesample/di/modules/AppModule.kt
package com.icostel.arhitecturesample.di.modules

import android.content.Context
import com.icostel.arhitecturesample.SampleApp
import com.icostel.arhitecturesample.di.factory_modules.ViewModelFactoryModule
import com.icostel.arhitecturesample.di.factory_modules.WorkManagerFactoryModule
import dagger.Binds
import dagger.Module

@Module(includes = [
    ViewModelFactoryModule::class,
    WorkManagerFactoryModule::class,
    RxWorkerModule::class,
    ViewModelModule::class,
    ApiModule::class,
    PreferenceModule::class,
    HttpClientModule::class,
    DbModule::class,
    UseCaseModule::class,
    ManagerModule::class,
    RepoModule::class,
    UtilsModule::class
])
abstract class AppModule {

    @Binds
    abstract fun bindContext(application: SampleApp): Context
}
